yaml2sbml/YamlModel.py

```diff
@@ -0,0 +1,589 @@
+
+        Returns:
+
+        Raises:
+            FileExistsError
+        """
+        # Check file ending.
+        if not (sbml_dir.endswith('.xml') or sbml_dir.endswith('.sbml')):
+            raise ValueError('sbml_dir should contain path to the sbml '
+                             'and hence end with .xml or .sbml')
+
+        if (not over_write) and os.path.exists(sbml_dir):
+            raise FileExistsError(f'Can not write SBML model. File {sbml_dir}'
+                                  f' already exists. Consider to set '
+                                  f'over_write=True.')
+
+        # generate SBML as string
+        sbml_as_string = _parse_yaml_dict(self._yaml_model)
+
+        with open(sbml_dir, 'w') as f_out:
+            f_out.write(sbml_as_string)
+
+    def write_to_petab(self,
+                       output_dir: str,
+                       model_name: str,
+                       petab_yaml_name: str = None,
+                       measurement_table_name: str = None):
+        """
+        Writes the YamlModel as a petab problem. Equivalent to calling yaml2petab
+        on the file produced by the yaml output.
+
+        If a petab_yaml_name is given, a .yaml file is created, that organizes
+        the petab problem. If additionally a measurement_table_file_name is
+        specified, this file name is written into the created .yaml file.
+
+        Arguments:
+            output_dir: path the output file(s) are be written out
+            model_name: name of SBML model
+            petab_yaml_name: name of yaml organizing the PEtab problem.
+            measurement_table_name: Name of measurement table
+        """
+        _yaml2petab(self._yaml_model,
+                    output_dir,
+                    model_name,
+                    petab_yaml_name,
+                    measurement_table_name)
+
+    def validate_model(self):
+        """
+        Validates the yaml model.
+
+        Arguments:
+
+        Returns:
+
+        Raises:
+            ValidationError
+        """
+        _validate_yaml_from_dict(self._yaml_model)
+
+    # functionalities regarding the time
+    def is_set_time(self):
+        return 'time' in self._yaml_model.keys()
+
+    def set_time(self,
+                 time_variable: str):
+        self._yaml_model['time'] = [time_variable]
+
+    def get_time(self):
+        return self._yaml_model['time'][0]
+
+    # functions adding a value
+    def add_parameter(self,
+                      parameter_id: str,
+                      over_write: bool = False,
+                      nominal_value: float = None,
+                      parameter_name: str = None,
+                      parameter_scale: str = None,
+                      lower_bound: float = None,
+                      upper_bound: float = None,
+                      estimate: float = None):
+        """
+        Adds a parameter. Overwrites an existing parameter with the same id,
+        if over_write=True.
```

Comment:
am i missing something or are the arguments, return, etc missing in the docstring?
same for most methods below.

Reply:
I left them out, since they should be similar, to what the corresponding block has as keys and hence the docstring would often be longer than the function. Should I add them?

Reply:
good point, the argument names are fairly self-explanatory. i don't think it's worth adding them just for the sake of consistency :)